src/db/oracledb/books-dao.js

```diff
@@ -0,0 +1,130 @@
+import _ from 'lodash';
+
+import oracledb from 'oracledb';
+
+import { parseQuery } from 'utils/parse-query';
+
+import { getConnection } from './connection';
+
+
+
+/**
+ * Return a list of books
+ *
+ * @param {object} query Query parameters
+ * @returns {Promise<object>[]} Promise object represents a list of books
+ */
+const getBooks = async (query) => {
+  const connection = await getConnection();
+  try {
+    const parsedQuery = parseQuery(query); // contrib is changed
```

Comment:
I would do this in the router level.

Reply:
Would be needing more explanation on this?

Reply:
Would be done inside GET by filters.